---
identifier: "CVE-2021-29570"
identifiers:
- "GHSA-545v-42p7-98fq"
- "CVE-2021-29570"
package_slug: "pypi/tensorflow-gpu"
title: "Out-of-bounds Read"
description: "TensorFlow is an end-to-end open source platform for machine learning.
  The implementation of `tf.raw_ops.MaxPoolGradWithArgmax` can cause reads outside
  of bounds of heap allocated data if attacker supplies specially crafted inputs.
  The implementation
  uses the same value to index in two different arrays but there is no guarantee that
  the sizes are identical. The fix will be included in TensorFlow 2.5.0. We will also
  cherrypick this commit on TensorFlow 2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and
  TensorFlow 2.1.4, as these are also affected and still in supported range."
date: "2021-05-21"
pubdate: "2021-05-21"
affected_range: "<2.1.4||>=2.2.0,<2.2.3||>=2.3.0,<2.3.3||>=2.4.0,<2.4.2"
fixed_versions:
- "2.1.4"
- "2.2.3"
- "2.3.3"
- "2.4.2"
affected_versions: "All versions before 2.1.4, all versions starting from 2.2.0 before
  2.2.3, all versions starting from 2.3.0 before 2.3.3, all versions starting from
  2.4.0 before 2.4.2"
not_impacted: "All versions starting from 2.1.4 before 2.2.0, all versions starting
  from 2.2.3 before 2.3.0, all versions starting from 2.3.3 before 2.4.0, all versions
  starting from 2.4.2"
solution: "Upgrade to versions 2.1.4, 2.2.3, 2.3.3, 2.4.2 or above."
urls:
- "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-545v-42p7-98fq"
- "https://nvd.nist.gov/vuln/detail/CVE-2021-29570"
- "https://github.com/tensorflow/tensorflow/commit/dcd7867de0fea4b72a2b34bd41eb74548dc23886"
- "https://github.com/advisories/GHSA-545v-42p7-98fq"
cvss_v2: "AV:L/AC:L/Au:N/C:P/I:N/A:P"
cvss_v3: "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:N/A:H"
uuid: "6999c083-3313-4625-9580-e00d07d161cc"
cwe_ids:
- "CWE-1035"
- "CWE-125"
- "CWE-937"
